### PR TITLE
88613 split create vs edit dtos for participants

### PR DESCRIPTION
**Pull request by janid1967**
[AB#88613](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/88613)

Refactor to avoid using Participant DTO's in Create context vs Edit context. In Edit context we need ID and RowVersion of entity to edit. In Create context we don't have ID/RowVersion. Removed constructors for such DTO's which had default value null for ID / RowVersion.

Also renamed phrase Person in Participant context, to avoid confusion between Persons in Person table. In Participant context we now name a person as an InvitedPerson.

### CODE REVIEW

File: src/Equinor.ProCoSys.IPO.Command/InvitationCommands/EditInvitation/EditInvitationCommandHandler.cs [hunk 1]
```diff
@@ -174,33 +174,33 @@ private async Task<List<CommPkg>> GetCommPkgScopeAsync(IList<string> commPkgNos,
...
-            IList<ParticipantsForCommand> participantsToUpdate,
+            IList<ParticipantsForEditCommand> participantsToUpdate,
             Invitation invitation)
         {
             var existingParticipants = invitation.Participants.ToList();
 
             var functionalRoleParticipants =
-                participantsToUpdate.Where(p => p.FunctionalRole != null).Select(p => p).ToList();
-            var functionalRoleParticipantIds = functionalRoleParticipants.Select(p => p.FunctionalRole.Id).ToList();
+                participantsToUpdate.Where(p => p.InvitedFunctionalRoleToEdit != null).Select(p => p).ToList();
```
Comment (TordJoranger) [inline src/Equinor.ProCoSys.IPO.Command/InvitationCommands/EditInvitation/EditInvitationCommandHandler.cs:183]: is the second select needed here?

Comment (janid1967) [reply src/Equinor.ProCoSys.IPO.Command/InvitationCommands/EditInvitation/EditInvitationCommandHandler.cs:183]: .Select(p => p) ?

File: src/Equinor.ProCoSys.IPO.Command/InvitationCommands/EditInvitation/EditInvitationCommandValidator.cs [hunk 2]
```diff
@@ -52,11 +52,15 @@ public EditInvitationCommandValidator(IInvitationValidator invitationValidator,
 
             RuleForEach(command => command.UpdatedParticipants)
                 .MustAsync((command, participant, _, cancellationToken) => ParticipantToBeUpdatedMustExist(participant, command.InvitationId, cancellationToken))
-                .WithMessage((command, participant) =>
-                    $"Participant with ID does not exist on invitation! Participant={participant}")
+                .WithMessage(_ => $"Participant with ID does not exist on invitation!")
+                .Must(participant => participant.SortKey >= 0)
```
Comment (TordJoranger) [inline src/Equinor.ProCoSys.IPO.Command/InvitationCommands/EditInvitation/EditInvitationCommandValidator.cs:56]: Maybe stupid question, but isn't the test in the code for SortKey > 1?

**Review by TordJoranger — APPROVED** [review]

File: src/Equinor.ProCoSys.IPO.Command/InvitationCommands/EditInvitation/EditInvitationCommandHandler.cs (hunk 1 above)
Comment (TordJoranger) [reply src/Equinor.ProCoSys.IPO.Command/InvitationCommands/EditInvitation/EditInvitationCommandHandler.cs]: yes, it seems to just copy the list once more. Unless we need new references(Deep copy), I don't see the point.

File: src/Equinor.ProCoSys.IPO.Command/InvitationCommands/EditInvitation/EditInvitationCommandValidator.cs (hunk 2 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.IPO.Command/InvitationCommands/EditInvitation/EditInvitationCommandValidator.cs:56]: The check in Handler for SortKey > 1 must be something else. 

I've just moved the rule from src/Equinor.ProCoSys.IPO.Command/InvitationCommands/ParticipantsForCommandValidator.cs which is now deleted, becauase the rule was never validated. The writer of code has misunderstood when Validators in FluentValidation kick in.

**Review by terjewoldseth — APPROVED** [review]

**Pull request merged**